sbpy/bib/core.py

```diff
@@ -29,8 +29,26 @@ def register(task, citations):
     """
     global _bibliography, _track
     if _track:
-        _bibliography[task] = _bibliography.get(task, {})
-        _bibliography[task].update(citations)
+        if task in _bibliography:
+            key = []
+            value = set()
+            key.append(list(citations.keys()))
+            for i in range(len(key)):
+                if list(citations.keys())[i] in _bibliography[task]:
+                    subtasks = _bibliography[task]
+                    if isinstance(list(subtasks.values())[i], list):
+                        for j in range(len(list(subtasks.values())[i])):
+                            value.add(list(subtasks.values())[i][j])
+                    else:
+                        value.add(list(subtasks.values())[i])
+                        value.add(list(citations.values())[i])
+                        subtasks[list(citations.keys())[i]] = value
+                else:
+                    subtasks = _bibliography[task]
+                    value.add(list(citations.values())[i])
+                    subtasks[list(citations.keys())[i]] = value
+        else:
+            _bibliography[task] = citations
```

Comment:
This part is a bit tricky: the value in `citation` is simply a string, but it should be a set. I couldn't find a better way to turn it into a set than this one:
`_bibliography[task] = {list(citation.keys())[0]: set(list(citation.values())[0])}`
Maybe you can find a better solution?